core/src/main/scala/cats/syntax/applicativeError.scala

```diff
@@ -3,12 +3,18 @@ package syntax
 
 import cats.data.{Xor, XorT}
 
-trait ApplicativeErrorSyntax {
+trait ApplicativeErrorSyntax1 {
+  implicit def applicativeErrorSyntax[F[_], E, A](fa: F[A])(implicit F: ApplicativeError[F, E]): ApplicativeErrorOps[F, E, A] =
+    new ApplicativeErrorOps[F, E, A](fa)
+}
+
+trait ApplicativeErrorSyntax extends ApplicativeErrorSyntax1 {
   implicit def applicativeErrorIdSyntax[E](e: E): ApplicativeErrorIdOps[E] =
     new ApplicativeErrorIdOps(e)
 
-  implicit def applicativeErrorSyntax[F[_, _], E, A](fa: F[E, A])(implicit F: ApplicativeError[F[E, ?], E]): ApplicativeErrorOps[F[E, ?], E, A] =
+  implicit def applicativeErrorSyntax2[F[_, _], E, A](fa: F[E, A])(implicit F: ApplicativeError[F[E, ?], E]): ApplicativeErrorOps[F[E, ?], E, A] =
```

Comment:
hmm.. actually I couldn't see a way for `Unapply` to work here as well. In `Unapply` the actual type `TC` only takes one type parameter `M` (which is a high-kinded type itself), while in our case the actual type is `ApplicativeError[F, E]`, which takes not only a highkinded type parameter, but also another type parameter `E`. So I think `Unapply` as is won't help here, and if I am right, we either keep this limited workaround (it's just a syntax helper, so I don't really see more complicated inference needed most of the cases), or we just do as you said, get rid of the support for `F[_, _]` and let people use Si-2712 fix for that.